code/__HELPERS/_lists.dm

```diff
@@ -146,23 +146,15 @@
 			return TRUE
 	return FALSE
 
-
-//Empties the list by setting the length to 0. Hopefully the elements get garbage collected
-/proc/clearlist(list/L)
-	if(!istype(L))
-		return
-
-	L.len = 0
-
-
-//Removes any null entries from the list
-/proc/listclearnulls(list/L)
-	if(!istype(L))
-		return
-
-	while(null in L)
-		L -= null
-
+/**
+ * Removes any null entries from the list
+ * Returns TRUE if the list had nulls, FALSE otherwise
+**/
+/proc/listclearnulls(list/list_to_clear)
+	var/start_len = length(list_to_clear)
+	var/list/new_list[start_len]
+	list_to_clear -= new_list
+	return length(list_to_clear) < start_len
```

Comment:
515 adds `list.RemoveAll()` which removes all instances of a given value in a list. I benchmarked this a while ago and it's much faster than listclearnulls.
```suggestion
#if DM_VERSION < 515
/proc/listclearnulls(list/list_to_clear)
	var/start_len = length(list_to_clear)
	var/list/new_list[start_len]
	list_to_clear -= new_list
	return length(list_to_clear) < start_len
#else
#define listclearnulls(list) list.RemoveAll(null)
#warn You can remove the old listclearnulls now
#endif
```